save-common/src/commonMain/kotlin/org/cqfn/save/core/files/FileUtils.kt

```diff
@@ -117,6 +118,30 @@ fun FileSystem.readFile(path: Path): String = this.read(path) {
     this.readUtf8()
 }
 
+/**
+ * Copies [source] with all files and subdirectories into [target]
+ *
+ * @param source a directory to copy
+ * @param target a destination directory
+ */
+fun FileSystem.copyRecursively(source: Path, target: Path) {
+    require(metadata(source).isDirectory)
```

Comment:
I remember, that we had some problems with `isDirectory` and should be accurate if there weren't updates in `okio`:

Some directories in unix systems can have extended permissions (like for example `/tmp`), which make their type not exactly `directory`, but `sticky directory`.

And `okio` for some reasons doesn't identify them as directory sometimes, however they are directories. 
It's strange, but `/tmp` in Linux and Mac is `sticky directory`, but `metadata(/tmp).isDirectory`  `== true` on Linux, but `== false` on Mac.

Previously for that reason we used `!isRegularFile`

Reply:
Thx, I keep forgetting about this. It's strange though, because `!isRegularFile` will be true not only for directories, but also for symlinks, sockets, pipes etc. And here 
https://github.com/square/okio/blob/ffbf472af20f39a18f46365c7a7dcf5a01f84cd1/okio/src/linuxX64Main/kotlin/okio/-LinuxX64PosixVariant.kt#L38
you can see that they check whether the file is a directory in an expected way.

Reply:
Changed, at least for now